src/Workspaces/Core/Portable/ProjectTelemetry/IRemoteProjectTelemetryService.cs

```diff
@@ -13,6 +14,20 @@ namespace Microsoft.CodeAnalysis.ProjectTelemetry
     /// </summary>
     internal interface IRemoteProjectTelemetryService
     {
-        ValueTask ComputeProjectTelemetryAsync(CancellationToken cancellation);
+        internal interface ICallback
+        {
+            ValueTask ReportProjectTelemetryDataAsync(RemoteServiceCallbackId callbackId, ProjectTelemetryData data, CancellationToken cancellationToken);
+        }
+
+        ValueTask ComputeProjectTelemetryAsync(RemoteServiceCallbackId callbackId, CancellationToken cancellation);
+    }
+
+    internal sealed class RemoteProjectTelemetryServiceCallbackDispatcher : RemoteServiceCallbackDispatcher, IRemoteProjectTelemetryService.ICallback
```

Comment:
💡 Separate file

Reply:
I'll keep these as they are for now. I'll follow up with more refactoring.